pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -1036,6 +1036,10 @@ func (m *kubeGenericRuntimeManager) GetPodStatus(uid kubetypes.UID, name, namesp
 	podIPs := []string{}
 	for idx, podSandboxID := range podSandboxIDs {
 		podSandboxStatus, err := m.runtimeService.PodSandboxStatus(podSandboxID)
+		if crierror.IsNotFound(err) {
```

Comment:
Same comment as above, needs tests AND a comment describing this as a race condition between the previous call and the current call.

Roughly, the previous call (getSandboxIDByPodUID) never fails due to a pod sandbox not existing.  Therefore, this method should not either, but instead act as if the previous call failed, which means the error should be ignored.